src/components/ApplicationListView/ApplicationListView.tsx

```diff
@@ -36,13 +40,35 @@ const ApplicationListView: React.FC<React.PropsWithChildren<unknown>> = () => {
   const applicationBreadcrumbs = useApplicationBreadcrumbs();
   const [canCreateApplication] = useAccessReviewForModel(ApplicationModel, 'create');
   const [canCreateComponent] = useAccessReviewForModel(ComponentModel, 'create');
+  const [nameFilter, setNameFilter] = useSearchParam('name', '');
+  const [onLoadName, setOnLoadName] = React.useState(nameFilter);
+
+  React.useEffect(() => {
+    if (nameFilter) {
+      setOnLoadName(nameFilter);
+    }
+  }, [nameFilter]);
 
   const [applications, loaded] = useApplications(namespace);
   applications?.sort(
     (app1, app2) =>
       +new Date(app2.metadata.creationTimestamp) - +new Date(app1.metadata.creationTimestamp),
   );
 
+  const filteredApplications = React.useMemo(() => {
+    return applications?.filter((app) => app.metadata.name.includes(nameFilter));
```

Comment:
If `app.spec.displayName` is available, filter list based on that else fallback on `app.metadata.name`.

Reply:
Added fallback and test